src/promptflow/promptflow/executor/_service/utils/process_utils.py

```diff
@@ -0,0 +1,115 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import asyncio
+import contextlib
+import json
+import multiprocessing
+import os
+from datetime import timedelta
+from multiprocessing import Queue
+from typing import Callable
+
+from promptflow._core._errors import UnexpectedError
+from promptflow._core.operation_context import OperationContext
+from promptflow._utils.exception_utils import ExceptionPresenter, JsonSerializedPromptflowException
+from promptflow._utils.logger_utils import service_logger
+from promptflow.exceptions import ErrorTarget
+from promptflow.executor._service._errors import ExecutionTimeoutError
+from promptflow.executor._service.utils.service_utils import get_log_context
+
+LONG_WAIT_TIMEOUT = timedelta(days=1).total_seconds()
+QUICK_TIMEOUT = 10  # seconds
+WAIT_SUBPROCESS_EXCEPTION_TIMEOUT = 10  # seconds
+
+
+async def invoke_function_in_process(
+    request, context_dict: dict, target_function: Callable, wait_timeout: int = LONG_WAIT_TIMEOUT
+):
+    with multiprocessing.Manager() as manager:
+        return_dict = manager.dict()
+        exception_queue = manager.Queue()
+
+        p = multiprocessing.Process(
+            target=execute_function,
+            args=(target_function, request, return_dict, exception_queue, context_dict),
+        )
+        p.start()
+        service_logger.info(f"[{os.getpid()}--{p.pid}] Start process to execute the request.")
+
+        # Wait for the process to finish or timeout
+        p.join(timeout=wait_timeout)
```

Comment:
This is a sync method, it will make the server hang

Reply:
Updated.